Geospatial/OGR/SpatialWriter.cs

```diff
@@ -0,0 +1,132 @@
+﻿using OSGeo.OGR;
+using OSGeo.OSR;
+using USACE.HEC.Results;
+
+namespace Geospatial.OGR;
+
+
+public class SpatialWriter : IResultsWriter
+{
+  private Layer _layer;
+  private DataSource _dataSource;
+  private SpatialReference _srs;
+  private SpatialReference _dst;
+  private bool _headersWritten;
+  private string _xField;
+  private string _yField;
+
+  /// <summary>
+  /// SpatialWriter writes spatial results data. X and Y Fields are the ResultItem names of the x and y coordinates. 
+  /// </summary>
+  public SpatialWriter(string outputPath, string driverName, int projection, string xField, string yField)
+  {
+    _dataSource = Ogr.GetDriverByName(driverName).CreateDataSource(outputPath, null) ?? throw new Exception("Failed to create data source.");
+    _srs = new SpatialReference("");
+    _srs.ImportFromEPSG(4326); // WGS84
+    if (_srs == null) throw new Exception("Failed to create SpatialReference.");
+    _dst = new SpatialReference("");
+    _dst.ImportFromEPSG(projection);
+    if (_dst == null) throw new Exception("Failed to create SpatialReference.");
+
+    _layer = _dataSource.CreateLayer("layer_name", _dst, wkbGeometryType.wkbPoint, null) ?? throw new Exception("Failed to create layer.");
+
+    _headersWritten = false;
+
+    _xField = xField;
+    _yField = yField;
+  }
+
+  public void Write(Result res)
+  {
+    if (!_headersWritten)
+    {
+      InitializeFields(res);
+      _headersWritten = true;
+    }
+
+    using var feature = new Feature(_layer.GetLayerDefn());
```

Comment:
Invert the naming. Put type on the left whenever possible. Same for the rest of this method.